### ISSUE TITLE
CSS file is not enqueued for posts outside the main loop

### ISSUE THREAD

**Issue by selul** [Codeinwp/gutenberg-blocks#308]
It seems that when we are on a singular context (`is_singular`) and we load content from posts other from the main post, the CSS file attached to that post is not enqueued as it should.

More details about this issue you can find here: https://github.com/Codeinwp/neve-pro-addon/issues/732

After a few debugging I've traced the issue to this line https://github.com/Codeinwp/gutenberg-blocks/blob/master/inc/css/class-block-frontend.php#L206 which seems to cause it. 

```
if ( is_singular() ) {
    return $this->enqueue_styles();
}
```
I think the main issue is that we return early here rather than still hooking to the_content to enqueue the style for rest of posts that display the_content, something like: 
```
                if ( is_singular() ) {
			$this->enqueue_styles();
		}

		add_filter(
			'the_content',
			function ( $content ) {
				if ( $this->has_excerpt ) {
					return $content;
				}

				$post_id = get_the_ID();

				$this->enqueue_styles( $post_id, true );

				return $content;
			}
		);
```

Basically, when we are on a singular context and we have posts outside the main loop, the handler that enqueues style based on the_content is never attached.
